web/src/user_group_edit_members.ts

```diff
@@ -66,17 +75,82 @@ function format_member_list_elem(person: User): string {
     });
 }
 
+function format_subgroup_list_elem(group: UserGroup): string {
+    return render_user_group_subgroup_entry({
+        group_id: group.id,
+        display_value: group.name,
+        can_edit: settings_data.can_manage_user_group(current_group_id),
+    });
+}
+
+function build_group_member_matcher(query: string): (member: User | UserGroup) => boolean {
+    query = query.trim();
+
+    const termlets = query.toLowerCase().split(/\s+/);
+    const termlet_matchers = termlets.map((termlet) => people.build_termlet_matcher(termlet));
+
+    return function (member: User | UserGroup): boolean {
+        if ("user_id" in member) {
+            const email = member.email.toLowerCase();
+
+            if (email.startsWith(query)) {
+                return true;
+            }
+
+            return termlet_matchers.every((matcher) => matcher(member));
+        }
+
+        const group_name = member.name;
+        if (group_name.startsWith(query)) {
+            return true;
+        }
+        return false;
+    };
+}
+
+function sort_group_member_email(a: User | UserGroup, b: User | UserGroup): number {
+    if ("user_id" in a && "user_id" in b) {
```

Comment:
Inintially had these conditions as `a.user_id !== undefined && b.user_id !== undefined` but linter gave an error.

Reply:
This is a little more precise than the original; ultimately the check is whether this is a User or UserGroup.